src/wordladder.cpp

```diff
@@ -267,31 +225,34 @@ void getWordLadder(const Lexicon &dictionary, string wordOne, string wordTwo) {
 }
 
 /*
- * Function: findNeighbourWords
+ * Function: findPartialWordLadders
  * Usage:
  * Find all the neighbour words from a given start word, which is a word of the same length but
- * differs from the start word by exactly 1 letter e.g. date and data
+ * differs from the start word by exactly 1 letter e.g. date and data, and create partial word
+ * ladders using this information in order to find final word ladder
  * Params: dictionary (Lexicon), usedWords (Set), queue (Queue of Stacks of strings),
  * firstLadder (Stack of strings)
  * -------------------------------------------------------------------------------------------------
  * Returns: None. Void function
 */
-void findNeighbourWords(
-        const Lexicon &dictionary
-        , Set<string> &usedWords
-        , Queue<Stack<string>> &queue
-        , const Stack<string> &firstLadder) {
+void findPartialWordLadders(
+        const Lexicon &dictionary,
+        Set<string> &usedWords,
+        Queue<Stack<string>> &queue,
+        const Stack<string> &firstLadder) {
+
     // Get the start word at the top of the initial word ladder
     string startWord = firstLadder.peek();
     // The algorith uses 2 nested loops for efficiency: 1st loop for index of the start word,
     // 2nd loop to loop through a-z of the alphabet
     for (int i = 0; i < startWord.length(); i++) {
-        for (char l = 'a'; l<='z'; ++l) {
+        for (char letter = 'a'; letter <='z'; ++letter) {

```

Comment:
Tiny thing that I missed in the first review: change this to `letter++` instead of `++letter`. Doing `++letter` will increment the value of letter _before_ each iteration of the for loop, but what you really want is the opposite: you want letter to be incremented _after_ each iteration of the for loop.

Reply:
How weird. This must be some kind of typo or copy and paste error because I'm confident that I intended letter++ 😕 anyway this is fixed!